pkg/tfgen/convert_cli.go

```diff
@@ -0,0 +1,453 @@
+//
+// This may need to be coarse-grain parallelized to speed up larger providers at the cost of more
+// memory, for example run 4 instances of `pulumi convert` on 25% of examples each.
+func (*cliConverter) convertViaPulumiCLI(
+	examples map[string]string,
+	mappings []tfbridge.ProviderInfo,
+) (
+	output map[string]translatedExample,
+	finalError error,
+) {
+	outDir, err := os.MkdirTemp("", "bridge-examples-output")
+	if err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to create a temp dir "+
+			" bridge-examples-output: %w", err)
+	}
+	defer func() {
+		if err := os.RemoveAll(outDir); err != nil {
+			if finalError == nil {
+				finalError = fmt.Errorf("convertViaPulumiCLI: failed to clean up "+
+					"temp bridge-examples-output dir: %w", err)
+			}
+		}
+	}()
+
+	examplesJSON, err := os.CreateTemp("", "bridge-examples.json")
+	if err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to create a temp "+
+			" bridge-examples.json file: %w", err)
+	}
+	defer func() {
+		if err := os.Remove(examplesJSON.Name()); err != nil {
+			if finalError == nil {
+				finalError = fmt.Errorf("convertViaPulumiCLI: failed to clean up "+
+					"temp bridge-examples.json file: %w", err)
+			}
+		}
+	}()
+
+	// Write example to bridge-examples.json.
+	examplesBytes, err := json.Marshal(examples)
+	if err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to marshal examples "+
+			"to JSON: %w", err)
+	}
+	if err := os.WriteFile(examplesJSON.Name(), examplesBytes, 0600); err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: failed to write a temp "+
+			"bridge-examples.json file: %w", err)
+	}
+
+	mappingsDir := filepath.Join(outDir, "mappings")
+
+	mappingsFile := func(i int, p tfbridge.ProviderInfo) string {
+		return filepath.Join(mappingsDir, fmt.Sprintf("%s.json", p.Name))
+	}
+
+	// Write out mappings files if necessary.
+	for i, m := range mappings {
+		if i == 0 {
+			if err := os.MkdirAll(mappingsDir, 0755); err != nil {
+				return nil, fmt.Errorf("convertViaPulumiCLI: failed to write "+
+					"mappings folder: %w", err)
+			}
+		}
+		copy := m
+		mpi := tfbridge.MarshalProviderInfo(&copy)
+		bytes, err := json.Marshal(mpi)
+		if err != nil {
+			return nil, fmt.Errorf("convertViaPulumiCLI: failed to write "+
+				"mappings folder: %w", err)
+		}
+		mf := mappingsFile(i, m)
+		if err := os.WriteFile(mf, bytes, 0600); err != nil {
+			return nil, fmt.Errorf("convertViaPulumiCLI: failed to write "+
+				"mappings file: %w", err)
+		}
+	}
+
+	pulumiPath, err := exec.LookPath("pulumi")
+	if err != nil {
+		return nil, fmt.Errorf("convertViaPulumiCLI: pulumi executalbe not "+
+			"in PATH: %w", err)
+	}
+
+	var mappingsArgs []string
+	for i, m := range mappings {
+		mappingsArgs = append(mappingsArgs, "--mappings", mappingsFile(i, m))
+	}
+
+	cmd := exec.Command(pulumiPath, append([]string{"convert",
+		"--from", "terraform",
+		"--language", "pcl",
+		"--out", outDir,
+		"--generate-only",
```

Comment:
`, "--", "--convert-examples", examplesJSON.Name()` should work here. What parse failure are you seeing when you try that?

Reply:
It doesn't work. I've tried. `pulumi convert` does not generate the output file.

Reply:
OK lemme check this out, I'm really not keen on us throwing env vars everywhere and this _should_ work.

Reply:
Yeh turns out it helps if you actually try to tell the flags library to parse the arguments 🤦 
I'll get this fixed up

Reply:
Works now. thank you.